pkg/drivers/qemu/qemu.go

```diff
@@ -515,13 +519,19 @@ func (d *Driver) Start() error {
 			time.Sleep(2 * time.Second)
 		}
 
-		if err != nil {
-			if isBootpdError(err) {
-				exit.Error(reason.IfBootpdFirewall, "ip not found", err)
-			}
+		if err == nil {
+			log.Debugf("IP: %s", d.IPAddress)
+			break
+		}
+		if !isBootpdError(err) {
 			return errors.Wrap(err, "IP address never found in dhcp leases file")
 		}
-		log.Debugf("IP: %s", d.IPAddress)
+		if unblockErr := firewall.UnblockBootpd(); unblockErr != nil {
+			klog.Errorf("failed unblocking bootpd from firewall: %v", unblockErr)
+			exit.Error(reason.IfBootpdFirewall, "ip not found", err)
```

Comment:
I am curious if we could return an error and let the CMD package to exit ? I generally frown upon exit in non-cmd code, but if there is a good reason or makes way cleaner code I am okay with Exit here.. 

what u thnk?

Reply:
The reason I did `exit.Error` is because it will retry if the first start fails. In the case where we know bootpd is being blocked and we failed to unblock the process it's almost certainly going to fail the retry so I was exiting. I could remove it,, the downside would be that it would have to go through the start loop one more time.